src/main.rs

```diff
@@ -116,14 +121,16 @@ fn main() {
         });
     }
 
-    let sleep_duration = std::time::Duration::new(config.run_interval, 0);
     let bitbucket = bitbucket::Bitbucket::new(&config.bitbucket, &fanout);
     if let Some(t) = config.clone().telegram {
         if t.enabled {
             t.announce_from(fanout.subscribe()).expect("Failed to authenticate with Telegram");
         }
     }
 
+    let schedule: CronSchedule = CronSchedule::parse(config.cron_interval).unwrap();
```

Comment:
With the combined `Config` field above, then we can handle a Cron schedule differently from a fixed interval.